assert.h conflicts with boost assert.h

Got this error message when trying to build with boost1.64 (linux,gcc7)
```
g++ -DHAVE_CONFIG_H -I. -I..  -D_GNU_SOURCE -D_REENTRANT -D_POSIX_C_SOURCE=200112L -D_POSIX_SOURCE -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=500 -Wno-long-long -Wall -fpermissive -D_FILE_OFFSET_BITS=64 -DFUSE_USE_VERSION=26 -DNDEBUG  -g -O2   -MT main.o -MD -MP -MF .deps/main.Tpo -c -o main.o main.cpp
In file included from /usr/include/boost/assert.hpp:58:0,
                 from /usr/include/boost/iostreams/detail/adapter/range_adapter.hpp:16,
                 from /usr/include/boost/iostreams/detail/push.hpp:19,
                 from /usr/include/boost/iostreams/filtering_stream.hpp:19,
                 from CompressionType.hpp:23,
                 from main.cpp:30:
/usr/include/boost/array.hpp: In member function ‘T& boost::array<T, N>::operator[](boost::array<T, N>::size_type)’
/usr/include/boost/array.hpp:118:20: error: expected primary-expression before ‘if’
             return BOOST_ASSERT_MSG( i < N, "out of range" ), elems[i];
                    ^
/usr/include/boost/array.hpp:118:20: error: expected ‘;’ before ‘if’
```
Looks like definition of assert() collide with boost assertion macros.
`# define BOOST_ASSERT_MSG(expr, msg) assert((expr)&&(msg))`
which in turns cause an error in operator[] in boost array.hpp
```
reference operator[](size_type i)
         {
             return BOOST_ASSERT_MSG( i < N, "out of range" ), elems[i];
         }
```

One solution would be to add `#define BOOST_DISABLE_ASSERTS` in `main.cpp, Compress.cpp` to disable boost assertion all together.

--- a/src/Compress.cpp
+++ b/src/Compress.cpp
@@ -27,6 +27,11 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 105600
+#define BOOST_DISABLE_ASSERTS
+#endif
+
 #include <boost/io/ios_state.hpp>
 
 #include <boost/scoped_array.hpp>

--- a/src/main.cpp
+++ b/src/main.cpp
@@ -29,6 +29,11 @@
 #include "FuseCompress.hpp"
 #include "CompressionType.hpp"
 
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 105600
+#define BOOST_DISABLE_ASSERTS
+#endif
+
 #include <boost/algorithm/string.hpp>
 #include <boost/program_options.hpp>
 #include <boost/tokenizer.hpp>

--- a/src/main_offline.cpp
+++ b/src/main_offline.cpp
@@ -37,6 +37,11 @@
 #include "CompressionType.hpp"
 #include "FileRememberXattrs.hpp"
 
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 105600
+#define BOOST_DISABLE_ASSERTS
+#endif
+
 #include <boost/scoped_array.hpp>
 #include <boost/algorithm/string.hpp>
 #include <boost/program_options.hpp>

--- a/src/tests/print_compress.cpp
+++ b/src/tests/print_compress.cpp
@@ -18,6 +18,11 @@
 #include "CompressionType.hpp"
 #include "FileManager.hpp"
 
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 105600
+#define BOOST_DISABLE_ASSERTS
+#endif
+
 #include <boost/scoped_array.hpp>
 #include <boost/algorithm/string.hpp>
 #include <boost/program_options.hpp>

